src/main/java/com/minecolonies/core/entity/other/cavalry/CavalryHorseEntity.java

```diff
@@ -0,0 +1,876 @@
+import net.minecraft.network.syncher.EntityDataAccessor;
+import net.minecraft.network.syncher.EntityDataSerializers;
+import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.DamageTypeTags;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityDimensions;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.Pose;
+import net.minecraft.world.entity.animal.Animal;
+import net.minecraft.world.entity.animal.horse.AbstractHorse;
+import net.minecraft.world.entity.animal.horse.Donkey;
+import net.minecraft.world.entity.animal.horse.Horse;
+import net.minecraft.world.entity.animal.horse.Mule;
+import net.minecraft.world.entity.animal.horse.Variant;
+import net.minecraft.world.entity.monster.Creeper;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.Vec3;
+import net.minecraft.world.entity.ai.attributes.AttributeInstance;
+import net.minecraft.world.entity.ai.attributes.Attributes;
+import net.minecraft.world.entity.ai.goal.BreedGoal;
+import net.minecraft.world.entity.ai.goal.FloatGoal;
+import net.minecraft.world.entity.ai.goal.FollowParentGoal;
+import net.minecraft.world.entity.ai.goal.LookAtPlayerGoal;
+import net.minecraft.world.entity.ai.goal.RandomLookAroundGoal;
+import net.minecraft.world.entity.ai.goal.RandomStandGoal;
+import net.minecraft.world.entity.ai.goal.WaterAvoidingRandomStrollGoal;
+import net.minecraft.world.entity.ai.navigation.PathNavigation;
+
+public class CavalryHorseEntity extends Horse
+{
+    private static final String RESERVE_KEY = "horse_reserved_for_cavalry";
+
+    public static final float BASE_W = 1.4F;
+    public static final float BASE_H = 1.6F;
+    public static final float SLIM_W = 0.70F;
+
+    private static final float SEATING_OFFSET = 0.40F;
+    public int logCooldown = 0;
+    public static final int LOG_COOLDOWN_INTERVAL = 200;
+
+    private static final int SLIM_GRACE_MAX_TICKS = 40;
+    private int slimGraceTicks = 0;
+
+    public static final float COMBAT_READINESS_THRESHOLD = .66f;
+
+    @Nullable
+    private BlockPos stablePos = null;
+    @Nullable
+    private ResourceKey<Level> stableDim = null;
+
+    private static final String NBT_STABLE_POS = "stablePos";
+    private static final String NBT_STABLE_DIM = "stableDim";
+    private static final String NBT_COMBAT_COOLDOWN = "combatCooldown";
+
+    private static final int RESERVATION_EXPIRATION_LIMIT = 200;
+    private int reservationExpiration = 0;
+
+    /*
+     * A measure of how unprepared for combat a mount is, representing damaged equipment, morale,
+     * fatigue or other non-HP factors.
+     */
+    private static final EntityDataAccessor<Float> DATA_COMBAT_CD =
```

Comment:
If possible, I'd avoid these. We should probably have a representation of the horse in the colony somewhere with a server/client side version that we sync ourselves manually.

We still have those in the citizens but I want to get rid of them for a while as they're quite slow to access and result in inconsistencies sometimes.